File: lib/string/urlJoin.ts
export function urlJoin(...args: any[]): string {
	if (!args) {
		return '';
	}

	const argLength = args.length;
	let urlResult = '';
	let joinCount = 0;

	for (let i = 0; i < argLength; i += 1) {
		if (args[i] !== null && args[i] !== undefined) {
			if (
				joinCount === 0 ||
				args[i].startsWith('/') ||
				args[i].startsWith('?') ||
				args[i].startsWith('&')
			) {
				urlResult += args[i];
			} else {
				urlResult += `/${args[i]}`;
			}

			joinCount += 1;
		}
	}

	return urlResult.replace(/\/$/g, '');
}

File: packages/python/qsu/string/urlJoin.py
import re


def urlJoin(*args) -> str:
	if not args:
		return ''

	urlResult = ''
	joinCount = 0

	for arg in args:
		if arg is not None:
			value = arg if isinstance(arg, type('')) else f'{arg}'

			if (
				joinCount == 0
				or value.startswith('/')
				or value.startswith('?')
				or value.startswith('&')
			):
				urlResult += value
			else:
				urlResult += '/' + value

			joinCount += 1

	return re.sub(r'/$', '', urlResult)
